### ISSUE TITLE
Caching the database queries

### ISSUE THREAD

**Issue by vikneswaran20** [Bhupesh-V/tutorialdb#37]
**Caching the DB queries will vastly improve the performance of this kind of projects**
This is a read-heavy web application where the read is pretty high more when comparted to write. As data becomes huge the query performance will be very slow. So introducing the caching will be good to scale up this application.

**Comment by Bhupesh-V** [comment]
Yes you are right @vikneswaran20 , I actually removed cached query from the latest release due to some problems
feel free to contribute and do let me know the approach :+1:

**Comment by vikneswaran20** [comment]
I have added the caching for the Tags page initially since the page is pretty static unless a new tag gets introduced. 
So I will be checking if there is a key in the cache for all the tags during tags page load if it doesn't I will query the DB and update the cache. Once the new tag has been added to the DB I will remove the cache. So during the next visit to tags page cache get automatically updated.
